File: Java/Ejercicio-2/Ejercicio-2-6/Ejercicio_2_6_4/src/ejercicio_2_6_4/Ejercicio_2_6_4.java
package ejercicio_2_6_4;

import java.util.Scanner;

/**
 * Clase Main para identificar si un número es primo usando un bucle for
 * @author daw1al13
 */
public class Ejepackage ejercicio_2_6_4;

import java.util.Scanner;

/**
 * Clase Main para identificar si un número es primo usando un bucle for sin break
 * @author daw1al13
 */
public class Ejercicio_2_6_4 {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in); // Scanner para introducir datos
        System.out.print("Dime un número: ");
        int numero = scan.nextInt(); // Valor del número introducido
        boolean esPrimo = true;      // Valor para determinar si es primo

        if (numero <= 1) {
            esPrimo = false; // 0 y 1 no son primos
        } else {
            // Bucle for que comprueba si el número es divisible por algún número menor que él
            for (int i = numero - 1; i > 1; i--) {
                if (numero % i == 0) {
                    esPrimo = false; // Si se puede dividir, no es primo
                }
            }
        }

        // Imprimir resultado
        if (esPrimo) {
            System.out.println("Tu número es primo");
        } else {
            System.out.println("Tu número no es primo");
        }
    }
}

File: Python/Ejercicio-2/Ejercicio-2-6/Ejercicio_2_6_4.py
"""
Clase Main para identificar si un número es primo usando un bucle for sin break
Autor: daw1al13
"""

def main():
    # Pedimos al usuario un número
    numero = int(input("Dime un número: "))
    
    # Valor para determinar si es primo
    es_primo = True

    if numero <= 1:
        es_primo = False  # 0 y 1 no son primos
    else:
        # Bucle for que comprueba si el número es divisible por algún número menor que él
        for i in range(numero - 1, 1, -1):
            if numero % i == 0:
                es_primo = False  # Si se puede dividir, no es primo

    # Imprimir resultado
    if es_primo:
        print("Tu número es primo")
    else:
        print("Tu número no es primo")


if __name__ == "__main__":
    main()
